crates/rspack_core/src/dependency/mod.rs

```diff
@@ -267,6 +221,10 @@ pub trait ModuleDependency: Dependency {
 }
 
 impl ModuleDependency for Box<dyn ModuleDependency> {
+  fn id(&self) -> &DependencyId {
```

Comment:
As far as I concerned, `id` should be available to all dependency types. So we'd better to move this to `Dependency`.

Reply:
Because we current only need id for module dependency, it is used generate module graph, but other dependency isn't need it and don't need creat id for it.

Reply:
Looks good to me.